### PR TITLE
HTTP API documents, part1

### PR DESCRIPTION
**Pull request by JinHai-CN**
### What problem does this PR solve?

1. dataset: create/delete/list/get/update
2. files in dataset: upload/download/list/delete/get_info

### Type of change

- [x] Documentation Update

### CODE REVIEW

File: api/http_api.md [hunk 1]
```diff
@@ -0,0 +1,758 @@
...
+- `"parse_mehtod"`: (*Body parameter*)  
+    Parsing method of the dataset.
+
+### Response
+
+The successful response includes a JSON object like the following:
+
+```shell
+{
+    "code": 0
```
Comment (yuzhichang) [inline api/http_api.md:64]: Need to unify `code` and `error_code`.

File: api/http_api.md [hunk 2]
```diff
@@ -0,0 +1,758 @@
...
+    "code": 3016,
+    "message": "Can't connect database"
+}
+```
+
+## Download a file from a dataset
+
+**GET** `/api/v1/dataset/{dataset_id}/documents/{document_id}`
+
+Uploads files to a dataset.
```
Comment (yuzhichang) [inline api/http_api.md]: `Uploads` -> `Downloads`

File: api/http_api.md [hunk 3]
```diff
@@ -0,0 +1,758 @@
...
+{
+    "code": 3016,
+    "message": "Can't connect database"
+}
+```
+
+
+## List files of a dataset
+
+**GET** `/api/v1/dataset/{dataset_id}/documents?keywords={keyword}&page={page}&page_size={limit}&orderby={orderby}&desc={desc}`
```
Comment (yuzhichang) [inline api/http_api.md]: No document for `page` and `page_size`

File: api/http_api.md [hunk 4]
```diff
@@ -0,0 +1,758 @@
...
+  - 'Authorization: Bearer {YOUR_ACCESS_TOKEN}'
+
+#### Request example
+
+```shell
+curl --request PUT \
+     --url http://{address}//api/v1/dataset/{dataset_id}/info \
+     --header 'Content-Type: application/json' \
+     --header 'Authorization: Bearer {YOUR_ACCESS_TOKEN}'
+     --raw '{
```
Comment (yuzhichang) [inline api/http_api.md:672]: It's better to move `document_id` to url

Comment (JinHai-CN) [reply api/http_api.md:672]: Fixed

File: api/http_api.md (hunk 3 above)
Comment (JinHai-CN) [reply api/http_api.md]: page and page_size is offset and limit to show in one page.

File: api/http_api.md (hunk 2 above)
Comment (JinHai-CN) [reply api/http_api.md]: Fixed

**Pull request merged**